src/internal/m365/graph/http_wrapper.go

```diff
@@ -99,15 +99,13 @@ func (hw httpWrapper) Request(
 
 	var resp *http.Response
 
-	i := 0
-
 	// stream errors from http/2 will fail before we reach
 	// client middleware handling, therefore we don't get to
 	// make use of the retry middleware.  This external
 	// retry wrapper is unsophisticated, but should only
 	// retry in the event of a `stream error`, which is not
 	// a common expectation.
-	for i < 3 {
+	for i := 0; i < 3; i++ {
```

Comment:
Do we need retries here, now that we are also retrying in graph adapter wrapper?
Could we move `streamErrRE` match logic to the wrapper?

Reply:
Sorry, nevermind. This won't be using the graph adapter.